compiler-plugin/src/main/kotlin/arrow/meta/quotes/WhenEntry.kt

```diff
@@ -0,0 +1,50 @@
+package arrow.meta.quotes
+
+import arrow.meta.Meta
+import arrow.meta.phases.ExtensionPhase
+import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtWhenCondition
+import org.jetbrains.kotlin.psi.KtWhenEntry
+
+/**
+ * A [KtWhenEntry] [Quote] with a custom template destructuring [WhenEntryScope]. See below:
+ *
+ * ```kotlin
```

Comment:
```suggestion
 * ```kotlin:ank:silent
```

Reply:
This is so it type-checks the snippets, otherwise the code would evolve separately from the docs as nobody goes back to change example unless they break the build

Reply:
Thank you for the explanation @raulraja , I've fixed that in the other PRs too